git-author:	Andrew Borodin <aborodin@…> (04/23/12 18:46:49)
git-committer:	Andrew Borodin <aborodin@…> (04/25/12 14:42:39)
Message:	Ticket #2785: error message when entering into compressed tar and cpio archives

Index: src/vfs/cpio/cpio.c
===================================================================
--- src/vfs/cpio/cpio.c	(revision 764cdf2497ac1fe79b4237e9c75d600f7d868351)
+++ src/vfs/cpio/cpio.c	(revision 8607bcc103ce2f9449b6b9736a9709cdb62a7178)
@@ -250,5 +250,5 @@
         mc_close (fd);
         s = g_strconcat (super->name, decompress_extension (type), (char *) NULL);
-        tmp_vpath = vfs_path_from_str (s);
+        tmp_vpath = vfs_path_from_str_flags (s, VPF_NO_CANON);
         fd = mc_open (tmp_vpath, O_RDONLY);
         vfs_path_free (tmp_vpath);
Index: src/vfs/tar/tar.c
===================================================================
--- src/vfs/tar/tar.c	(revision 764cdf2497ac1fe79b4237e9c75d600f7d868351)
+++ src/vfs/tar/tar.c	(revision 8607bcc103ce2f9449b6b9736a9709cdb62a7178)
@@ -317,5 +317,5 @@
         mc_close (result);
         s = g_strconcat (archive->name, decompress_extension (type), (char *) NULL);
-        tmp_vpath = vfs_path_from_str (s);
+        tmp_vpath = vfs_path_from_str_flags (s, VPF_NO_CANON);
         result = mc_open (tmp_vpath, O_RDONLY);
         vfs_path_free (tmp_vpath);
